### PR TITLE
[Feature] Implememnt the universal visualizer for multiple task.

### PR DESCRIPTION
**Pull request by mzr1996**
## Motivation

Rename the original `ClsVisualizer` to `UniversalVisualizer` to include more tasks.

## Modification

1. `ClsVisualizer` -> `UniversalVisualizer`
2. `ClsVisualizer.add_datasample` -> `UniversalVisualizer.visualize_cls`
3. The `name` argument is not necessary argument now.
4. Support `visualize_image_retrieval` and `visualize_masked_image`.
5. Implement the `visualize` method of the `ImageRetrievalInferencer`.

## BC-breaking

The `name` argument is not necessary argument now, and it's not the first argument of these visualization
methods. And the original `add_datasample` method is also renamed.

## Use cases

1. To visualize the image classification result.
```python
from mmpretrain.visualization import UniversalVisualizer
from mmpretrain.structures import DataSample
import mmcv

visualizer = UniversalVisualizer()
image = mmcv.imread('demo/bird.JPEG')[..., ::-1]  # The visualization methods accept RGB image.
data_sample = DataSample().set_gt_label(1).set_pred_label(2).set_pred_score([0., 0.8, 0.2])
visualizer.visualize_cls(image, data_sample, show=True)
```
visualize_cls

2. To visualize the image retrieval result
```python
from mmpretrain import ImageRetrievalInferencer

inferencer = ImageRetrievalInferencer('resnet50-arcface_8xb32_inshop', prototype='demo/')
inferencer('demo/bird.JPEG', topk=3, show=True)
```
img_retrieval

3. To visualize the masked image.
```python
from mmpretrain.visualization import UniversalVisualizer
from mmpretrain.structures import DataSample
import mmcv
import torch

visualizer = UniversalVisualizer()
image = mmcv.imread('demo/bird.JPEG')[..., ::-1]  # The visualization methods accept RGB image.
data_sample = DataSample().set_mask(torch.tensor([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]]))
visualizer.visualize_masked_image(image, data_sample, show=True)
```
masked_image

### CODE REVIEW

File: mmpretrain/visualization/visualizer.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        int: The adaptive scale.
+    """
+    short_edge_length = min(img_shape)
+    scale = short_edge_length / 224.
+    return min(max(scale, min_scale), max_scale)
+
+
+@VISUALIZERS.register_module()
+class UniversalVisualizer(Visualizer):
+    """Universal Visualizer for classification task.
```
Comment (fangyixiao18) [inline mmpretrain/visualization/visualizer.py:43]: not only classification task?

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: docs/en/useful_tools/cam_visualization.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## Introduction of the CAM visualization tool
 
-MMClassification provides `tools\visualizations\vis_cam.py` tool to visualize class activation map. Please use `pip install "grad-cam>=1.3.6"` command to install [pytorch-grad-cam](https://github.com/jacobgil/pytorch-grad-cam).
+MMClassification provides `tools/visualization/vis_cam.py` tool to visualize class activation map. Please use `pip install "grad-cam>=1.3.6"` command to install [pytorch-grad-cam](https://github.com/jacobgil/pytorch-grad-cam).
```
Comment (fangyixiao18) [inline docs/en/useful_tools/cam_visualization.md:5]:
```suggestion
MMPretrain provides `tools/visualization/vis_cam.py` tool to visualize class activation map. Please use `pip install "grad-cam>=1.3.6"` command to install [pytorch-grad-cam](https://github.com/jacobgil/pytorch-grad-cam).
```

File: docs/zh_CN/useful_tools/cam_visualization.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## 类别激活图可视化工具介绍
 
-MMClassification 提供 `tools\visualizations\vis_cam.py` 工具来可视化类别激活图。请使用 `pip install "grad-cam>=1.3.6"` 安装依赖的 [pytorch-grad-cam](https://github.com/jacobgil/pytorch-grad-cam)。
+MMClassification 提供 `tools/visualization/vis_cam.py` 工具来可视化类别激活图。请使用 `pip install "grad-cam>=1.3.6"` 安装依赖的 [pytorch-grad-cam](https://github.com/jacobgil/pytorch-grad-cam)。
```
Comment (fangyixiao18) [inline docs/zh_CN/useful_tools/cam_visualization.md:5]: MMPretrain

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Review by fangyixiao18 — APPROVED** [review]
